stellargraph/data/unsupervisedSampler.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Jan 16 15:00:01 2019
+
+@author: hab031
+"""
+
+
+__all__ = ["UnsupervisedSampler"]
+
+
+import random
+import numpy as np
+
+from stellargraph.core.utils import is_real_iterable
+from stellargraph.core.graph import StellarGraphBase
+from stellargraph.data.explorer import UniformRandomWalk
+
+
+class UnsupervisedSampler:
+    """
+        This class does unsupervised samlping in a graph. 
+        As the minimum, tt takes a Stellargraph graph and and walker instance. The walker is the the graph sampling strategy, such as uniform random walk, biased random walk etc.
+        At the moment only uniform random walks are performed but this can be easily modified to accomodate other types of walks.
+        Given a graph and walker, the generator method of this Unsupervisedsampler generates a batch of positive and negative samples. The generator method requires a batch size.
+        The positive samples are all the (target, context) pairs from the walks generated by walker. Negative samples are contexts generated for each target based on a sampling strategy.
+    """
+
+    def __init__(
+        self, G, walker=None, nodes=None, length=2, number_of_walks=1, seed=None
+    ):
+
+        """
+           Initializing all the paramters for the class instance. A Stellargraph is a must. 
+           The root nodes and walker are required. However, if not provided they default to all nodes in the graph and uniform random walker respectively.
```

Comment:
The documentation for a class should be under the class definition, not in `__init__` (this is so the API docs are generated properly)